eZ/Publish/API/Repository/Tests/ContentServiceTest.php

```diff
@@ -6186,4 +6186,143 @@ private function getExpectedMediaContentInfoProperties()
             'status' => ContentInfo::STATUS_PUBLISHED,
         ];
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\LocationCreateStruct[] $locationsToHide
+     *
+     * @dataProvider provideLocationsToHideAndReveal
+     */
+    public function testHideContent(array $locationsToHide)
+    {
+        $repository = $this->getRepository();
+
+        $contentTypeService = $repository->getContentTypeService();
+
+        $contentType = $contentTypeService->loadContentTypeByIdentifier('folder');
+
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+
+        $contentCreate = $contentService->newContentCreateStruct($contentType, 'eng-US');
+        $contentCreate->setField('name', 'Folder to hide');
+
+        $content = $contentService->createContent(
+            $contentCreate,
+            $locationsToHide
+        );
+
+        $publishedContent = $contentService->publishVersion($content->versionInfo);
+        $locations = $locationService->loadLocations($publishedContent->contentInfo);
+
+        // Sanity check
+        $this->assertCount(3, $locations);
+        $this->assertEquals(
+            [
+                false,
+                false,
+                false,
+            ],
+            array_column($locations, 'hidden')
+        );
+
+        /* BEGIN: Use Case */
+        $contentService->hideContent($publishedContent->contentInfo);
+        /* END: Use Case */
+
+        $hiddenLocations = $locationService->loadLocations($publishedContent->contentInfo);
+        $this->assertEquals(
+            [
+                true,
+                true,
+                true,
+            ],
+            array_column($hiddenLocations, 'hidden')
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\LocationCreateStruct[] $locationsToHide
+     *
+     * @dataProvider provideLocationsToHideAndReveal
+     */
+    public function testRevealContent(array $locationsToReveal)
```

Comment:
Hmm, I'd actually try to reuse previous test results via `@depends` and `$this->getRepository(false)`, but just nitpicking, so don't worry about it if you prefer it this way ;)